src/cpp/shared_core/include/shared_core/Error.hpp

```diff
@@ -709,6 +709,36 @@ Error systemError(
    const Error& in_cause,
    const ErrorLocation& in_location);
 
+/**
+ * @brief Function which creates a system error associated with a particular system call.
+ *
+ * @param in_function        The error message to associate with this error.
```

Comment:
I think the description here would be more accurate if it were something like `The system function for which this error occurred.`

Reply:
You're right -- I had forgotten to update the text here; thanks for catching that.